src/main/scala/com/preprocessing/edgeList/Compressor.scala

```diff
@@ -0,0 +1,177 @@
+    argStr += s"$outfile\t"
+    println(s"args: $argStr")
+
+    var resetIndex = 0
+
+    // a map between the original vertex id to the vertex id in the compressed, sorted representation
+    // TODO; map may contain large amount of vertices; can/should it be distributed?
+    // var permutationPairRDD = RDD[(Int, Int)]()?
+    val permutationMap = collection.mutable.Map[Int, Int]()
+
+    def readEdgeList(): RDD[(Int, Int)] = {
+      val distFile: RDD[String] = sc.textFile(infile)
+
+      val edgeList: RDD[(Int, Int)] = distFile
+        .filter(s => !s.contains("#")) // ignore comments
+        .map(s => {
+          val split = s.split(sep)
+          (split(0).toInt, split(1).toInt)
+        })
+
+
+      edgeList
+      // for some reason, issues with performing the deduplication at this stage
+      //        .filter(edge => edge._1 != edge._2)
+      //        .distinct()
+    }
+
+    def getOrElseAutoIncrement(permutationMap: collection.mutable.Map[Int, Int], key: Int): Unit = {
+      if (!permutationMap.isDefinedAt(key)) {
+        permutationMap(key) = resetIndex
+        resetIndex += 1
+      }
+    }
+
+    println("reading edge list...")
+    val edgeList = readEdgeList()
+
+    println("sorting edge list...")
+    val sortedEdgeList = edgeList.sortBy(e => (e._1, e._2))
+    val sorted = sortedEdgeList.collect()
+
+    println("mapping edge list...")
+    sorted.foreach(edge => {
+      getOrElseAutoIncrement(permutationMap, edge._1)
+    })
+    sorted.foreach(edge => {
+      getOrElseAutoIncrement(permutationMap, edge._2)
+    })
+
+    println("remapping edge list...")
+    val remapped = edgeList
+      .map(edge => {
+        (permutationMap(edge._1), permutationMap(edge._2))
+      })
+      // remove self directed edges and duplicate edges
+      .filter(edge => edge._1 != edge._2)
+      .distinct()
+      .sortBy(edge => (edge._1, edge._2))
+
+    println("ensuring number of edges, number of nodes is consistent")
+    val remappedNNodes: Int = permutationMap.size
+    val remappedNEdges: Int = remapped.count().toInt
+
+    println(s"Input listed ${nNodes} vertices; After compression: ${remappedNNodes}")
+    println(s"Input listed ${nEdges} edges; After compression: ${remappedNEdges}")
+    assert(remappedNNodes == nNodes)
+    //    assert(remappedNEdges == nEdges)
+
+    println(s"writing compressed edge list to $outfile...")
+    try {
+      remapped
+        .map(e => s"${e._1} ${e._2}")
+        .coalesce(1, false) // TODO; partition into multiple files/numPartitions HERE
+        .saveAsTextFile(outfile)
+    } catch {
+      case e: org.apache.hadoop.mapred.FileAlreadyExistsException => println("File already exists, please delete the existing file")
+
+    }
+
+    class VertexID(i: String) {
+      val id: String = i
+      println(id)
+      override def hashCode(): Int = {
+        println(id)
+        id.split("\\.")(1).toInt
+      }
+    }
+
+    val vid = new VertexID("9.1")
```

Comment:
I think this was scratch from our chat, so note to self to remove

Reply:
removed